src/main/java/fruitshop/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    private final Storage storage = new Storage();
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getQuantity() >= 0) {
+            storage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+        } else {
+            throw new RuntimeException("negative balance cannot be recorded at "
+                    + BalanceOperation.class);
+        }
+    }
```

Comment:
The `BalanceOperation` class should not instantiate `Storage` directly. This violates the Dependency Inversion principle of SOLID. Instead, consider passing a reference to `Storage` through the constructor or using a dependency injection framework.